logical_type.go

```diff
@@ -14,12 +14,16 @@ import (
 	"fmt"
 	"math"
 	"math/big"
+	"regexp"
+	"strings"
 	"time"
 )
 
 type toNativeFn func([]byte) (interface{}, []byte, error)
 type fromNativeFn func([]byte, interface{}) ([]byte, error)
 
+var m = make(map[string]*regexp.Regexp)
```

Comment:
For a long lived variable with global scope, I recommend using a name that gives it context, such as `reFromPattern`.